module/validation/receipt_validator_test.go

```diff
@@ -295,39 +299,258 @@ func (s *ReceiptValidationSuite) TestMultiReceiptInvalidParent() {
 	// assuming signatures are all good
 	s.verifier.On("Verify", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
 
-	prepareReceipts := func() []*flow.ExecutionReceipt {
-		// G <- A <- B <- C
-		blocks, result0, _ := unittest.ChainFixture(4)
-
-		receipts := unittest.ReceiptChainFor(blocks, result0)
-		blockA, blockB, blockC := blocks[1], blocks[2], blocks[3]
-		receiptA := receipts[1]
-		blockA.Payload.Receipts = []*flow.ExecutionReceipt{}
-		blockB.Payload.Receipts = []*flow.ExecutionReceipt{receiptA}
-		blockC.Payload.Receipts = []*flow.ExecutionReceipt{}
-		// update block header so that blocks are chained together
-		unittest.ReconnectBlocksAndReceipts(blocks, receipts)
-		// assuming all receipts are executed by the correct executor
-		for _, r := range receipts {
-			r.ExecutorID = s.ExeID
-		}
-
-		for _, b := range blocks {
-			s.Extend(b)
-		}
-		s.PersistedResults[result0.ID()] = result0
-		return receipts
+	// G <- A <- B <- C
+	blocks, result0, _ := unittest.ChainFixture(4)
+
+	receipts := unittest.ReceiptChainFor(blocks, result0)
+	blockA, blockB, blockC := blocks[1], blocks[2], blocks[3]
+	receiptA := receipts[1]
+	receiptBInvalid := receipts[2]
+	receiptC := receipts[3]
+	blockA.Payload.Receipts = []*flow.ExecutionReceiptMeta{}
+	blockB.Payload.Receipts = []*flow.ExecutionReceiptMeta{receiptA.Meta()}
+	blockB.Payload.Results = []*flow.ExecutionResult{&receiptA.ExecutionResult}
+	blockC.Payload.Receipts = []*flow.ExecutionReceiptMeta{}
+	// update block header so that blocks are chained together
+	unittest.ReconnectBlocksAndReceipts(blocks, receipts)
+	// assuming all receipts are executed by the correct executor
+	for _, r := range receipts {
+		r.ExecutorID = s.ExeID
+	}
+
+	for _, b := range blocks {
+		s.Extend(b)
+	}
+	s.PersistedResults[result0.ID()] = result0
+
+	candidate := unittest.BlockWithParentFixture(blockC.Header)
+	candidate.Payload = &flow.Payload{
+		Receipts: []*flow.ExecutionReceiptMeta{receiptBInvalid.Meta(), receiptC.Meta()},
+		Results:  []*flow.ExecutionResult{&receiptBInvalid.ExecutionResult, &receiptC.ExecutionResult},
 	}
 
-	receipts := prepareReceipts()
 	// make receipt B as bad
-	receipts[2].ExecutorID = s.VerID
-	// recieptB and receiptC
-	receiptsInNewBlock := []*flow.ExecutionReceipt{receipts[2], receipts[3]}
-	err := s.receiptValidator.Validate(receiptsInNewBlock)
+	receiptBInvalid.ExecutorID = s.VerID
+	// receiptB and receiptC
+	err := s.receiptValidator.ValidatePayload(&candidate)
 	s.Require().Error(err)
```

Comment:
better to also test the error type here:
```suggestion
	s.Require().Error(err)
	require.True(s.T(), engine.IsInvalidInputError(err), err)	
```